corehq/apps/linked_domain/tests/test_linked_userreports.py

```diff
@@ -141,14 +141,27 @@ def test_remote_link_ucr(self, fake_ucr_getter):
 
         api_key, _ = HQApiKey.objects.get_or_create(user=django_user)
         auth_headers = {'HTTP_AUTHORIZATION': 'apikey test:%s' % api_key.key}
-        self.domain_link.save()
 
-        url = reverse('linked_domain:ucr_config', args=[self.domain, self.report.get_id])
-        headers = auth_headers.copy()
-        headers[REMOTE_REQUESTER_HEADER] = self.domain_link.linked_domain
-        resp = self.client.get(url, **headers)
+        # mark domain link to be remote
+        self.domain_link.remote_base_url = "http://find/my/app"
 
-        fake_ucr_getter.return_value = json.loads(resp.content)
+        def _reset_domain_link():
+            self.domain_link.remote_base_url = None
+        self.addCleanup(_reset_domain_link)
+
+        def _get_response():
+            url = reverse('linked_domain:ucr_config', args=[self.domain, self.report.get_id])
+            headers = auth_headers.copy()
+            headers[REMOTE_REQUESTER_HEADER] = self.domain_link.linked_domain
+            resp = self.client.get(url, **headers)
+
+            response_json = resp.json()
+            return {
+                "report": ReportConfiguration.wrap(response_json["report"]),
+                "datasource": DataSourceConfiguration.wrap(response_json["datasource"]),
+            }
```

Comment:
would all this be necessary if you just extracted the view code from `'linked_domain:ucr_config'` into a utility function and used that.

I guess it's not clear to me exactly what is being tested here. You could have a test for the view itself and a separate test for updating the linked domain with data 'from the view'.